crates/cheatcodes/src/utils.rs

```diff
@@ -172,6 +173,47 @@ pub(super) fn sign(private_key: &U256, digest: &B256) -> Result {
     Ok((v, r, s).abi_encode())
 }
 
+pub(super) fn sign_with_wallet<DB: DatabaseExt>(
+    ccx: &mut CheatsCtxt<DB>,
+    signer: Option<Address>,
+    digest: &B256,
+) -> Result {
+    if let Some(script_wallets) = &ccx.state.script_wallets {
+        let mut script_wallets = script_wallets.inner.lock();
+        let maybe_provided_sender = script_wallets.provided_sender;
+        let signers = script_wallets.multi_wallet.signers()?;
+
+        let signer = if let Some(signer) = signer {
+            signer
+        } else if let Some(provided_sender) = maybe_provided_sender {
+            provided_sender
+        } else if signers.len() == 1 {
+            *signers.keys().next().unwrap()
+        } else {
+            return Err("could not determine signer".into());
+        };
+
+        let wallet = signers
+            .get(&signer)
+            .ok_or_else(|| fmt_err!("signer with address {signer} is not available"))?;
+
+        let sig = RuntimeOrHandle::new()
+            .block_on(wallet.sign_hash(digest))
+            .map_err(|err| fmt_err!("{err}"))?;
+
+        let recovered = sig.recover(digest.to_ethers()).map_err(|err| fmt_err!("{err}"))?;
+        assert_eq!(recovered.to_alloy(), signer);
+
+        let v = U256::from(sig.v);
```

Comment:
Should `v` be like above? note that there is a `to_u64` which does all of `sig.v().y_parity_byte_non_eip155().unwrap_or(sig.v().y_parity_byte())`

Reply:
this `sig` is still from ethers and we've just returned a `sig.v` earlier

https://github.com/foundry-rs/foundry/blame/ae0ef191bb9fe144af6736bf56a4fa25f781d39a/crates/cheatcodes/src/utils.rs#L172

Looking at `sig.v().y_parity_byte_non_eip155().unwrap_or(sig.v().y_parity_byte()` I am wondering if this is correct actually

Isn't ecrecover expects `v` to be 27 or 28? and here we might end up with 0 or 1